docs_src/optimizations.md

```diff
@@ -324,6 +324,23 @@ additional selector which is the logical NOR of all of the original `Select`
 selector bits is appended to the `OneHotSelect` selector and the respective case
 is the value selected when all selector bits are zero (`case_0` in the diagram).
 
+### Simplifying select arms
+
+Within any given arm of a `Select` multiplexer, it is safe to assume that the
+selector has the specific value required to select that arm.
+
+![drawing](./selector_substitution.png)
+
+This makes it possible to try simplifying the arms based on the selector
```

Comment:
I think a key compilery term to include here is "specialization". So maybe "This makes it possible to specialize the arms ... "

Reply:
Nice!